File: 프로그래머스/2/42586. 기능개발/기능개발.py
# 프로그래머스 42586 기능 개발 LV.2
def solution(progresses, speeds):
    from math import ceil
    answer = []
    days = [ceil((100 - p) / q) for p, q in zip(progresses, speeds)]
    stack = []
    for day in days:
        if len(answer) == 0 or stack[-1] < day:
            answer.append(1)
            stack.append(day)
        else:
            answer[-1] += 1
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    Deque<Integer> stack = new ArrayDeque<>();
    
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new LinkedList<>();
        for (int i = 0; i < speeds.length; i++) {
            int due = (int) Math.ceil((100 - (double) progresses[i]) / (double) speeds[i]);
            
            if (stack.isEmpty() || stack.peekLast() < due) {
                stack.addLast(due);
                answer.add(1);
                continue;
            }
            
            int tmp = answer.get(answer.size() - 1);
            answer.remove(answer.size() - 1);
            answer.add(tmp + 1);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}